lib/rules/no-unknown-arguments-for-builtin-components.js

```diff
@@ -0,0 +1,158 @@
+  let query = pureQuery;
+  let fuzzyResults = [];
+
+  while (!fuzzyResults.length && query.length) {
+    fuzzyResults = filter(candidates, query);
+    query = query.slice(0, -1);
+  }
+
+  const msg = `"${argumentName}" is unknown argument for <${tagName} /> component.`;
+  if (fuzzyResults.length) {
+    return `${msg} Did you mean "@${fuzzyResults[0]}"?`;
+  } else {
+    return msg;
+  }
+}
+
+function REQUIRED_MESSAGE(tagName, argumentNames) {
+  return `Argument${argumentNames.length > 1 ? 's' : ''} ${argumentNames
+    .map((el) => `"@${el}"`)
+    .join(' or ')} is required for <${tagName} /> component.`;
+}
+
+function CONFLICT_MESSAGE(argumentName, rawList) {
+  const conflictsList = rawList.filter((el) => `@${el}` !== argumentName);
+  return `"${argumentName}" conflicts with ${conflictsList
+    .map((el) => `"@${el}"`)
+    .join(', ')}, only one should exists.`;
+}
+
+function isArgument(attributeNode) {
+  return attributeNode.name.charAt(0) === '@';
+}
+
+function pureName(attributeNode) {
+  return attributeNode.name.replace('@', '');
+}
+
+module.exports = class NoUnknownArgumentsForBuiltinComponents extends Rule {
+  visitor() {
+    return {
+      ElementNode(node) {
+        if (!(node.tag in KnownArguments)) {
+          return;
+        }
+
+        let nodeMeta = KnownArguments[node.tag];
+        let warns = [];
+        let seen = [];
+
+        const logError = (attr) => {
+          this.log({
+            message: ERROR_MESSAGE(node.tag, attr.name),
+            line: attr.loc && attr.loc.start.line,
+            column: attr.loc && attr.loc.start.column,
+            source: (this.sourceForNode(attr) || '').split('=')[0],
+          });
+        };
+
+        const logConflict = (attr, conflictList) => {
+          this.log({
+            message: CONFLICT_MESSAGE(attr.name, conflictList),
+            line: attr.loc && attr.loc.start.line,
+            column: attr.loc && attr.loc.start.column,
+            source: (this.sourceForNode(attr) || '').split('=')[0],
+          });
+        };
+
+        const logRequired = (variants) => {
+          this.log({
+            message: REQUIRED_MESSAGE(node.tag, variants),
+            line: node.loc && node.loc.start.line,
+            column: node.loc && node.loc.start.column + 1,
+            source: node.tag,
+          });
+        };
+
+        for (let argument of node.attributes) {
+          if (!isArgument(argument)) {
+            continue;
+          }
+          const argumentName = pureName(argument);
+          if (!nodeMeta.arguments.includes(argumentName)) {
+            warns.push(argument);
+          } else {
+            seen.push(argumentName);
+          }
+        }
+
+        for (let warn of warns) {
+          logError(warn);
+        }
+
+        if ('conflicts' in nodeMeta) {
+          for (let conflictList of nodeMeta.conflicts) {
+            if (conflictList.every((item) => seen.includes(item))) {
```

Comment:
Shouldn't this be `some`?

Reply:
nope, we have array of conflicted items `[model, models]`, and if both exists - we emit error, if only one exist - there is no conflict

Reply:
to be honest I don't like `for of` syntax, here, `forEach` is more readable, but it's prohibited in https://github.com/ember-template-lint/ember-template-lint/pull/1708